projects/packages/search/src/class-module-control.php

```diff
@@ -148,23 +148,49 @@ public function deactivate() {
 		 */
 		do_action( 'jetpack_deactivate_module_' . self::JETPACK_SEARCH_MODULE_SLUG );
 
+		$this->disable_instant_search();
+
 		return $success;
 	}
 
+	/**
+	 * Update module status
+	 *
+	 * @param boolean $active - true to activate, false to deactivate.
+	 */
+	public function update_status( $active ) {
+		return $active ? $this->activate() : $this->deactivate();
+	}
+
 	/**
 	 * Disable Instant Search Experience
 	 */
 	public function disable_instant_search() {
+		if ( ! $this->is_instant_search_enabled() ) {
+			return true;
+		}
```

Comment:
Why is this check necessary? What's the benefit of bypassing the `update_option` call here?

Reply:
The only benefit is on some condition, we could use less write operations. But I guess you are right, no need to do so, the situation is pretty rare. No need to do this.

Reply:
removed in 78aae3f1e7